src/anomalib/metrics/spro.py

```diff
@@ -0,0 +1,214 @@
+            separated by mask files.
+        kwargs: Additional arguments to the TorchMetrics base class.
+
+    Example:
+        Import the metric from the package:
+
+        >>> import torch
+        >>> from anomalib.metrics import SPRO
+
+        Create random ``preds`` and ``labels`` tensors:
+
+        >>> labels = torch.randint(low=0, high=2, size=(2, 10, 5), dtype=torch.float32)
+        >>> labels = [labels]
+        >>> preds = torch.rand_like(labels[0][:1])
+
+        Compute the SPRO score for labels and preds:
+
+        >>> spro = SPRO(threshold=0.5)
+        >>> spro.update(preds, labels)
+        >>> spro.compute()
+        tensor(0.6333)
+
+        .. note::
+            Note that the example above shows random predictions and labels.
+            Therefore, the SPRO score above may not be reproducible.
+
+    """
+
+    def __init__(self, threshold: float = 0.5, saturation_config: str | Path | None = None, **kwargs) -> None:
+        super().__init__(**kwargs)
+        self.threshold = threshold
+        self.saturation_config = load_saturation_config(saturation_config) if saturation_config is not None else None
+        if self.saturation_config is None:
+            logger.warning(
+                "The saturation_config attribute is empty, the threshold is set to the defect area."
+                "This is equivalent to PRO metric but with the 'region' are separated by mask files",
+            )
+        self.add_state("score", default=torch.tensor(0.0), dist_reduce_fx="sum")
+        self.add_state("total", default=torch.tensor(0), dist_reduce_fx="sum")
+
+    def update(self, predictions: torch.Tensor, masks: list[torch.Tensor]) -> None:
+        """Compute the SPRO score for the current batch.
+
+        Args:
+            predictions (torch.Tensor): Predicted anomaly masks.
+            masks (list[torch.Tensor]): Ground truth anomaly masks with original height and width. Each element in the
+                list is a tensor list of masks for the corresponding image.
+
+        Example:
+            To update the metric state for the current batch, use the ``update`` method:
+
+            >>> spro.update(preds, labels)
+        """
+        score, total = spro_score(
+            predictions,
+            masks,
+            threshold=self.threshold,
+            saturation_config=self.saturation_config,
+        )
+        self.score += score
+        self.total += total
+
+    def compute(self) -> torch.Tensor:
+        """Compute the macro average of the SPRO score across all masks in all batches.
+
+        Example:
+            To compute the metric based on the state accumulated from multiple batches, use the ``compute`` method:
+
+            >>> spro.compute()
+            tensor(0.5433)
+        """
+        if self.total == 0:  # only background/normal images
+            return torch.Tensor([1.0])
+        return self.score / self.total
+
+
+def spro_score(
```

Comment:
```suggestion
def compute_spro_score(
```

Reply:
Not sure about this one. I think `spro_score` is in line with TorchMetrics naming convention, where the functional metrics carry the name of the metric (so without `compute`).